### PR TITLE
Learning rate scheduler for language model

### PR DESCRIPTION
**Pull request by emilydinan**
1. Adds a function "receive_metrics" for Agent class which allows the agent to receive a dictionary
2. Calls this function after validation during the train_model loop
3. Implements this function in language_model so that the learning rate is decreased by a factor when the validation loss does not decrease. Defaults to 0.5
4. Also makes small change to "get_predictions" function in language_model so that it handles case when predictions are empty (this happens when there are no 'valid' examples in the batch)

### CODE REVIEW

File: parlai/agents/language_model/language_model.py [hunk 1]
```diff
@@ -487,6 +494,15 @@ def shutdown(self):
...
         super().shutdown()
 
+    def receive_metrics(self, metrics_dict):
+        if 'loss' in metrics_dict:
+            if self.best_val_loss is None:
+                self.best_val_loss = metrics_dict['loss']
+            else:
+                if metrics_dict['loss'] > self.best_val_loss:
+                    self.lr *= self.lr_factor
+                    print("Updating learning rate: lr =", self.lr)
```
Comment (alexholdenmiller) [inline parlai/agents/language_model/language_model.py]: do you want to use loss or something like `opt.get('valid_metric', opt.get('accuracy'))`?

Comment (alexholdenmiller) [reply parlai/agents/language_model/language_model.py]: nevermind
